OpenStack-Rabbit-Consumer/usr/local/bin/aq_api.py

```diff
@@ -82,8 +91,150 @@ def aq_manage(hostname, env_type, env_name):
+    s.mount('https://', HTTPAdapter(max_retries=retries))
+
+    response = s.delete(url, auth=HTTPKerberosAuth())
+    machinename = response.text
+    if response.status_code != 200:
+        logger.error("Aquilon delete machine failed: " + str(response.text))
+        logger.error(url)
+        raise Exception("Aquilon machine delete failed")
+
+    logger.info("Successfully deleted machine")
+
+
+def host_create(hostname, machinename,firstip,archetype,domain,personality,osname,osversion):
+    logger.info("Attempting to create host for %s " % (hostname))
+
+    url = common.config.get("aquilon", "url") + ADD_HOST_SUFFIX.format(hostname,machinename,firstip,archetype,domain,personality,osname,osversion)
+
+    verify_kerberos_ticket()
+
+    s = requests.Session()
+    s.verify = "/etc/grid-security/certificates/"
+    retries = Retry(total=5, backoff_factor=0.1, status_forcelist=[503])
+    s.mount('https://', HTTPAdapter(max_retries=retries))
+
+    response = s.put(url, auth=HTTPKerberosAuth())
+    if response.status_code != 200:
+        logger.error("Aquilon create host failed: " + str(response.text))
+        logger.error(url)
+        raise Exception("Aquilon host create failed")
+
+    logger.info("Successfully created host")
+
+def host_delete(hostname):
+    logger.info("Attempting to delete host for %s " % (hostname))
+
+    url = common.config.get("aquilon", "url") + DELETE_HOST_SUFFIX.format(hostname)
+
+    verify_kerberos_ticket()
+
+    s = requests.Session()
+    s.verify = "/etc/grid-security/certificates/"
+    retries = Retry(total=5, backoff_factor=0.1, status_forcelist=[503])
+    s.mount('https://', HTTPAdapter(max_retries=retries))
+
+    response = s.delete(url, auth=HTTPKerberosAuth())
+    if response.status_code != 200:
+        logger.error("Aquilon delete host failed: " + str(response.text))
+        logger.error(url)
+        raise Exception("Aquilon host delete failed")
+
+    logger.info("Successfully deleted host")
+
+
+
+def add_machine_interface(machinename,ipaddr,macaddr,label,interfacename,hostname):
+    logger.info("Attempting to add ip %s to machine %s " % (ipaddr,machinename))
+
+    url = common.config.get("aquilon", "url") + ADD_INTERFACE_SUFFIX.format(machinename,interfacename,macaddr)
+
+    verify_kerberos_ticket()
+
+    s = requests.Session()
+    s.verify = "/etc/grid-security/certificates/"
+    retries = Retry(total=5, backoff_factor=0.1, status_forcelist=[503])
+    s.mount('https://', HTTPAdapter(max_retries=retries))
+
+    response = s.put(url, auth=HTTPKerberosAuth())
+    if response.status_code != 200:
+        logger.error("Aquilon add interface failed: " + str(response.text))
+        logger.error(url)
+        raise Exception("Aquilon add interface failed")
+def add_machine_interface_address(machinename,ipaddr,macaddr,label,interfacename,hostname):
+    logger.info("Attempting to add address ip %s to machine %s " % (ipaddr,machinename))
```

Comment:
There seems to be a lot of repetition in these functions. could it be factored out?